crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/discover.rs

```diff
@@ -261,12 +295,11 @@ pub(crate) fn get_cached_remotes_near_basis(
     async move {
         let mut nodes = Vec::new();
 
-        let query = QueryAgentsEvt::new(inner.space.clone())
+        let query = QueryAgentsEvt::new(inner.space())
             .near_basis(basis_loc)
             .limit(LIMIT);
-        for node in inner.evt_sender.query_agents(query).await? {
+        for node in inner.query_agents(query).await? {
             if !inner
-                .i_s
                 .is_agent_local(node.agent.clone())
                 .await
                 .unwrap_or(true)
```

Comment:
This unwrap isn't covered. I guess this `is_agent_local` call would fail if the sender had an error but the check could also fail I supposed. It would be good to describe the behavior here as filtering out agents if we can't check. 

It's probably some other code's job to notice that the sender is closed so it's fair enough that we don't check for a closed error here.

Reply:
After some consideration, I decided to actually this error up instead of converting it to a true. This should still only occur around shutdown time.

Reply:
also added a test for that

Reply:
That makes sense, and the test looks good